File: SecureBackend/AES_CBC.js
'use strict';

const crypto = require('crypto');

var AESalgoProv = 'aes-128-cbc';
const IV= Buffer.from([0x15, 0x14, 0x13, 0x12, 0x11,
            0x10, 0x09, 0x08, 0x07, 0x06, 0x05, 0x04, 0x03, 0x02, 0x01, 0x00]);
            
exports.AesEncryption = function(data, key) {
    var encodeKey = crypto.createHash('md5').update(new Buffer(key), 'utf-8').digest();
    var cipher = crypto.createCipheriv(AESalgoProv, encodeKey, IV);
    var encrypted = cipher.update(data, 'utf-8', 'base64');
    encrypted += cipher.final('base64');
    return encrypted;
};
exports.AesDecryption = function(encryptedata, key) {
    var encodeKey = crypto.createHash('md5').update(new Buffer(key), 'utf-8').digest();
    var decipher = crypto.createDecipheriv(AESalgoProv, encodeKey, IV);
    var decoded  = decipher.update(encryptedata,'base64', 'utf8');

    decoded += decipher.final('utf8');
    return decoded;
};

File: PythonClient/Ciphers/Symmetric/AesCBC.py
import base64
import hashlib
from array import array

from Crypto.Cipher import AES

import pkcs7
from AesCiphers import Ciphers


class CBC(Ciphers):
    def __init__(self):
        iv = [0x15, 0x14, 0x13, 0x12, 0x11, 0x10, 0x09, 0x08, 0x07, 0x06, 0x05, 0x04, 0x03, 0x02, 0x01, 0x00]
        self.IV = array('b', iv)

    def Encrypt(self, PlainText, SecurePassword):
        pw_encode = SecurePassword.encode('utf-8')
        text_encode = PlainText.encode('utf-8')

        key = hashlib.md5(pw_encode).digest()

        cipher = AES.new(key, AES.MODE_CBC, self.IV)
        pad_text = pkcs7.encode(text_encode)
        msg = cipher.encrypt(pad_text)

        EncodeMsg = base64.b64encode(msg)
        return EncodeMsg

    def Decrypt(self, Encrypted, SecurePassword):
        decodbase64 = base64.b64decode(Encrypted.decode("utf-8"))
        pw_encode = SecurePassword.decode('utf-8')

        key = hashlib.md5(pw_encode).digest()

        cipher = AES.new(key, AES.MODE_CBC, self.IV)
        msg = cipher.decrypt(decodbase64)
        pad_text = pkcs7.decode(msg)

        decryptedString = pad_text.decode('utf-8')
        return decryptedString
